platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JdkTools.java

```diff
@@ -50,87 +49,47 @@
 import java.util.function.Consumer;
 import java.util.function.Function;
 
-import static java.lang.ClassLoader.getSystemClassLoader;
-
 /**
  * Subset replacement for {@link javax.tools.ToolProvider} that avoids the application class loader.
  */
 public class JdkTools {
 
     // Copied from ToolProvider.defaultJavaCompilerName
     private static final String DEFAULT_COMPILER_IMPL_NAME = "com.sun.tools.javac.api.JavacTool";
+    private static final String DEFAULT_CONTEXT_IMPL_NAME = "com.sun.tools.javac.util.Context";
 
     private final ClassLoader isolatedToolsLoader;
-    private final boolean isJava9Compatible;
 
     private Class<JavaCompiler.CompilationTask> incrementalCompileTaskClass;
 
-    JdkTools(Jvm jvm, List<File> compilerPlugins) {
+    JdkTools(List<File> compilerPlugins) {
         DefaultClassLoaderFactory defaultClassLoaderFactory = new DefaultClassLoaderFactory();
-        JavaVersion javaVersion = jvm.getJavaVersion();
-        boolean java9Compatible = javaVersion.isJava9Compatible();
         ClassLoader filteringClassLoader = getSystemFilteringClassLoader(defaultClassLoaderFactory);
-        if (!java9Compatible) {
-            File toolsJar = jvm.getToolsJar();
-            if (toolsJar == null) {
-                throw new IllegalStateException("Could not find tools.jar. Please check that "
-                    + jvm.getJavaHome().getAbsolutePath()
-                    + " contains a valid JDK installation.");
```

Comment:
💭 The reason why this is gone is that the check moved up higher to https://github.com/gradle/gradle/pull/28281/files#diff-e56e568a70f371e8fa937e65f7cb5673480123d1ce9665b18114aa84c9a81686R85

❓ I think I should replace the rather short exception there with this one.